pkg/backup/backup/backup_manager.go

```diff
@@ -694,10 +693,10 @@ func (bm *backupManager) saveClusterMetaToExternalStorage(b *v1alpha1.Backup, cs
 	return "", nil
 }
 
-func (bm *backupManager) tryBackupIfCanSnapshot(b *v1alpha1.Backup, tc *v1alpha1.TidbCluster) (string, error) {
+func (bm *backupManager) volSnapshotBackup(b *v1alpha1.Backup, tc *v1alpha1.TidbCluster) (string, error) {
 	if s, reason, err := snapshotter.NewSnapshotterForBackup(b.Spec.Mode, bm.deps); err != nil {
 		return reason, err
-	} else if s != nil {
+	} else if s != (&snapshotter.NoneSnapshotter{}) {
```

Comment:
```suggestion
	} else if _, ok := s.(*snapshotter.NoneSnapshotter); !ok {
```

I think we can use this judgement, or keep `if s != nil`. `s != (&snapshotter.NoneSnapshotter{})` is a wrong method, maybe it compares their address, and it always gets `false`, even if `s := &snapshotter.NoneSnapshotter{}`. In go, we can't compare two objects directly, use `reflect.DeepEqual()` is a choice.

Reply:
fix in https://github.com/pingcap/tidb-operator/pull/4784/commits/286dee8f3243b0f3a3f6d3b5f1f927d09a601993